system-addon/lib/SearchFeed.jsm

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ /* globals ContentSearch, XPCOMUtils, Services */
+"use strict";
+
+const {utils: Cu} = Components;
+const {actionTypes: at, actionCreators: ac} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+const SEARCH_ENGINE_TOPIC = "browser-search-engine-modified";
+const CONTENT_SEARCH_READY = "content-search-initialized";
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "ContentSearch",
+  "resource:///modules/ContentSearch.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+this.SearchFeed = class SearchFeed {
+  addObservers() {
+    Services.obs.addObserver(this, SEARCH_ENGINE_TOPIC, false);
+    Services.obs.addObserver(this, CONTENT_SEARCH_READY, false);
+  }
+  removeObservers() {
+    Services.obs.removeObserver(this, SEARCH_ENGINE_TOPIC);
+    Services.obs.removeObserver(this, CONTENT_SEARCH_READY);
+  }
+  observe(subject, topic, data) {
+    switch (topic) {
+      case CONTENT_SEARCH_READY:
+        this.getState();
```

Comment:
So here we can listen to ```browser-ui-startup-complete``` and it will work in the case that we get initialized before ContentSearch gets initialized from nsBrowserGlue, but in the case where ContentSearch.init happens first and we haven't had time to set up our observer for that notification yet, we still need to check, on our init, if ContentSearch is ready to be used. So we still need to expose a ```isInit``` value in ContentSearch.